### PR TITLE
PXN-4209 Tracks Alignment

### PR DESCRIPTION
**Pull request by Esteban-ML**
## What have changed?
Some tracks were no showing up correctly in Melidata, due to the implementation of 3 similar VM's.
* PXPaymentCongratsViewModel
* PXResultViewModel
* PXBusinessResultViewModel

Only the PXPaymentCongratsViewModel was working, and the remaining VM's were later transformed into a PXPaymentCongratsViewModel, loosing some important information for the analytics.

image

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/PXPaymentCongratsViewModel.swift [hunk 1]
```diff
@@ -308,6 +320,10 @@ extension PXPaymentCongratsViewModel: PXNewResultViewModelInterface {
 
     // CALLBACKS & TRACKING
     func getTrackingProperties() -> [String: Any] {
+        if let track = paymentCongrats.trackingData?.getTrackingProperties() {
+            return track
+        }
+
         if let internalTrackingValues = paymentCongrats.internalTrackingValues {
```
Comment (jlainog-meli) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/PXPaymentCongratsViewModel.swift:327]: what are this `internalTrackingValues` is it possible is the same we are just doing in this PR?

**Review by jlainog-meli — APPROVED** [review]
LGTM, just a minor question.

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/PXPaymentCongratsViewModel.swift (hunk 1 above)
Comment (Esteban-ML) [reply MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/PXPaymentCongratsViewModel.swift:327]: This is different, our `trackingData` will provide everything to make a successful track but this `internalTrackingValues` property will only provide the properties/info of a track.

**Review by jlainog-meli — APPROVED** [review]

**Review by rgrajalesmeli — APPROVED** [review]
LG!

**Comment by jlainog-meli** [comment]
migrated

**Pull request closed without merging**